app/bundles/LeadBundle/Segment/Query/QueryBuilder.php

```diff
@@ -200,20 +201,19 @@ public function getState()
     /**
      * Executes this query using the bound parameters and their types.
      *
-     * Uses {@see Connection::executeQuery} for select statements and {@see Connection::executeUpdate}
-     * for insert, update and delete statements.
+     * @return Result<mixed>|int|string
      *
-     * @return \Doctrine\DBAL\Driver\Statement|int
-     *
-     * @throws \Doctrine\DBAL\DBALException
+     * @throws \Exception
      */
     public function execute()
     {
-        if (self::SELECT == $this->type) {
-            return $this->connection->executeQuery($this->getSQL(), $this->params, $this->paramTypes);
-        } else {
-            return $this->connection->executeUpdate($this->getSQL(), $this->params, $this->paramTypes);
+        if (self::SELECT === $this->type) {
```

Comment:
@escopecz I had to restore this function but I am using the latest version of the code, the cause of this action is `$this->params, $this->paramTypes` are private properties in this class and they are not visible to the parent. I think this file needs restructuring